libs/hiraya/src/components/Button/style.ts

```diff
@@ -9,27 +9,31 @@ interface ButtonProps {
   buttonSizes?: keyof typeof buttonSizeStyle;
   isFullWidth?: boolean;
   icon?: ReactNode;
+  focus?: boolean;
+  disabled?: boolean;
 }
 
 const ButtonStyled = styled(MUIButton)<ButtonProps>(
-  ({ variants, buttonSizes, isFullWidth, icon }) => ({
-    backgroundColor: variantStyle[variants || 'primary'].background,
-    color: variantStyle[variants || 'primary'].text,
-    padding: '0',
+  ({ variants, buttonSizes, isFullWidth, icon, focus, disabled }) => ({
+    ...buttonSizeStyle[buttonSizes || 'large'],
 
     '&.MuiButton-root': {
-      minWidth: icon
-        ? buttonSizeStyle[buttonSizes || 'large']
-        : isFullWidth
-        ? '112px'
-        : '92px',
-      height: buttonSizeStyle[buttonSizes || 'large'],
+      width: isFullWidth ? '100%' : buttonSizeStyle[buttonSizes || 'large'],
     },
 
+    ...(disabled
+      ? { ...disabledState }
+      : { ...variantStyle[variants || 'primary'] }),
+
     borderRadius: '5px',
-    border: variantStyle[variants || 'primary'].border,
+
     cursor: 'pointer',
     transition: 'background-color 0.3s ease',
+
+    '&:focus': {
+      boxShadow: '0 0 10px 5px #a9b7ea',
+      backgroundColor: variantStyle[variants || 'primary'].hoverBackground,
+    },
```

Comment:
Maybe you can add the &:focus as part of the variantStyles like this

```js
const variantStyle = {
  primary: {
    background: '#305EFF',
    color: 'white',
    border: 'none',
    hoverBackground: '#2950DA',
    '&:focus': {
      boxShadow: '0 0 10px 5px #a9b7ea',
      backgroundColor: "primary color"
    },
  },
  // rest of variantStyles here with &:focus prop
```

Then we can remove this specific part so it doesn't apply to disabled state